internal/impl/gcp/input_spanner.go

```diff
@@ -0,0 +1,187 @@
+	ProjectID         string
+	InstanceID        string
+	DatabaseID        string
+	StreamID          string
+	StartTimestamp    time.Time
+	EndTimestamp      time.Time
+	HeartbeatInterval time.Duration
+}
+
+func parseRFC3339Nano(pConf *service.ParsedConfig, key string) (time.Time, error) {
+	s, err := pConf.FieldString(key)
+	if err != nil {
+		return time.Time{}, err
+	}
+	if s == "" {
+		return time.Time{}, nil
+	}
+
+	t, err := time.Parse(time.RFC3339Nano, s)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("failed to parse %v as RFC3339Nano: %w", key, err)
+	}
+	return t, nil
+}
+
+func parseDuration(pConf *service.ParsedConfig, key string) (time.Duration, error) {
+	s, err := pConf.FieldString(key)
+	if err != nil {
+		return 0, err
+	}
+	if s == "" {
+		return 0, nil
+	}
+
+	d, err := time.ParseDuration(s)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse %v as duration: %w", key, err)
+	}
+	return d, nil
+}
+
+func spannerInputConfigFromParsed(pConf *service.ParsedConfig) (conf spannerInputConfig, err error) {
+	if conf.ProjectID, err = pConf.FieldString(siFieldProjectID); err != nil {
+		return
+	}
+	if conf.InstanceID, err = pConf.FieldString(siFieldInstanceID); err != nil {
+		return
+	}
+	if conf.DatabaseID, err = pConf.FieldString(siFieldDatabaseID); err != nil {
+		return
+	}
+	if conf.StreamID, err = pConf.FieldString(siFieldStreamID); err != nil {
+		return
+	}
+	if conf.StartTimestamp, err = parseRFC3339Nano(pConf, siFieldStartTimestamp); err != nil {
+		return
+	}
+	if conf.EndTimestamp, err = parseRFC3339Nano(pConf, siFieldEndTimestamp); err != nil {
+		return
+	}
+	if conf.HeartbeatInterval, err = parseDuration(pConf, siFieldHeartbeatInterval); err != nil {
+		return
+	}
+	return
+}
+
+func spannerInputSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Beta().
+		Version("TODO").
+		Categories("Services", "GCP").
+		Summary("Creates an input that consumes from a spanner change stream.").
+		Field(service.NewStringField(siFieldProjectID).Description("GCP project ID containing the Spanner instance")).
+		Field(service.NewStringField(siFieldInstanceID).Description("Spanner instance ID")).
+		Field(service.NewStringField(siFieldDatabaseID).Description("Spanner database ID")).
+		Field(service.NewStringField(siFieldStreamID).Description("The name of the change stream to track")).
+		Field(service.NewStringField(siFieldStartTimestamp).Optional().Description("RFC3339 formatted timestamp to start reading from (default: current time)").Default("")).
+		Field(service.NewStringField(siFieldEndTimestamp).Optional().Description("RFC3339 formatted timestamp to stop reading at (default: no end time)").Default("")).
+		Field(service.NewStringField(siFieldHeartbeatInterval).Optional().Description("Duration string for heartbeat interval (e.g., '10s')").Default("10s")).
+		Field(service.NewAutoRetryNacksToggleField())
+}
+
+func init() {
+	err := service.RegisterInput("gcp_spanner", spannerInputSpec(),
```

Comment:
```suggestion
	err := service.RegisterInput("gcp_spanner_cdc", spannerInputSpec(),
```

Having `cdc` in the name is important IMO to distinguish that it is CDC because people might just search for CDC

Reply:
Renamed types, files, registration and corrected commit messages